### ISSUE TITLE
CrossEntropyLoss with all zero targets

### ISSUE THREAD

**Issue by edervishaj** [Jyonn/Legommenders#18]
Hi,

When training with negative sampling, the loss is computed as the cross entropy loss for both the positive and negative candidate items. However, the targets are set to zero for all candidates:

https://github.com/Jyonn/Legommenders/blob/033032cab7739f7ba51451e28cd8208990b6378a/model/legommender.py#L137-L139

Is this the expected behavior?

Thanks!

**Comment by Jyonn** [comment]
Hi, these lines of codes are correct. The shape of labels is (batch_size), indicating that the positive item index is 0, not representing all the candidates.

**Comment by Jyonn** [comment]
The related processing code can be found here:

https://github.com/Jyonn/Legommenders/blob/033032cab7739f7ba51451e28cd8208990b6378a/loader/resampler.py#L72-L77

**Comment by edervishaj** [comment]
I was under the impression that for each of the candidates in `scores`, you wanted to provide the label click vs. non-click.

Thanks, closing this.
